config_system/log_handlers.py

```diff
@@ -74,3 +75,28 @@ def errors(self):
 
     def criticals(self):
         return self.counts["CRITICAL"]
+
+
+class ColorFormatter(logging.Formatter):
+    """Formatter that provide colored messages for console logging when possible"""
+
+    colors = {
+        "WARNING": 3,
+        "CRITICAL": 1,
+        "ERROR": 1,
+        "DEBUG": 6,
+        "INFO": 2
+    }
+    color_fmt = "\033[1;3{}m"
+    reset = "\033[0m"
+
+    def __init__(self, fmt, enabled=False):
+        super(ColorFormatter, self).__init__(fmt)
+        self.enabled = enabled
+
+    def format(self, log_msg):
+        paint = self.colors.get(log_msg.levelname)
+        if paint and self.enabled:
+            log_msg = deepcopy(log_msg)
+            log_msg.levelname = self.color_fmt.format(paint) + log_msg.levelname + self.reset
```

Comment:
Because `colors`, `color_fmt` and `reset` aren't member variables, would it be more correct to refer to them as `ColorFormatter.colors` etc rather than via `self`?

Reply:
They are ColorFormatter class attributes, thus `self` is the proper way to access them.

Reply:
Class variables (attributes) can be accessed either via self (`self.color_fmt`) or the class name (`ColorFormatter.color_fmt`).

If we use `self` for access it's not clear at the point we are using the variable that this is a class variable, and we could unintentionally re-bind the variable (by assigning to it).

If we use the class name for access then a sub-class of `ColorFormatter` could not override the variable.

I think it makes sense to access `color_fmt` and `reset` via the class names - a subclass should never have to override these.

That leaves `colors`. I suggest that we initialise `colors` as an instance variable in `__init__`. In the future we might wither want to subclass `ColorFormatter`, or provide functions to modify log coloring i.e. `setColor()`

Reply:
Such separation can be achieved with private class attribute.
Changed as requested.